community/logging/src/main/java/org/neo4j/logging/RotatingFileOutputStreamSupplier.java

```diff
@@ -190,40 +231,58 @@ private void rotate()
             return;
         }
 
-        Runnable runnable = () -> {
-            OutputStream newStream;
-            try
+        Runnable runnable = () ->
+        {
+            synchronized ( outRef )
```

Comment:
Maybe we can use `ReadWriteLock` instead of synchronized, write lock here and in `#close()` and read lock in all other places?